flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -204,13 +208,8 @@ public ClusterID getClusterId() {
 		return clusterClientFactory.createClusterDescriptor(executorConfig);
 	}
 
-	public EnvironmentInstance createEnvironmentInstance() {
-		try {
-			return wrapClassLoader(EnvironmentInstance::new);
-		} catch (Throwable t) {
-			// catch everything such that a wrong environment does not affect invocations
-			throw new SqlExecutionException("Could not create environment instance.", t);
-		}
+	public EnvironmentInstance getEnvironmentInstance() {
```

Comment:
Actually, I'm not sure whether this simple change is the right way to go. 

IIUC you are trying to make `LocalExecutor` stateful, right? I think we might need to first change the design of current SQL CLI, get the relations between `LocalExecutor`, `SessionContext`, `ExecutionContext` and `EnvironmentInstance` clear. 

I've create a jira to track this issue: https://issues.apache.org/jira/browse/FLINK-14672

I'd prefer to finish that issue first and then come back to this one.

Reply:
IMO, the sessionContext only have parts of the info about the SQL session, properties and views but have no tables  and catalogs, functions etc.  the ExecutionContext have all of the things, so in theory, the state we need to save is `ExecutionContext`.   For EnvironmentInstance, it's a wrapper of ExecutionContext.